ddtrace/mocktracer/mockspan.go

```diff
@@ -7,282 +7,122 @@ package mocktracer // import "github.com/DataDog/dd-trace-go/v2/ddtrace/mocktrac
 	}
-	s.Lock()
-	defer s.Unlock()
-	if s.finished {
-		return
+	for k, v := range s.Metrics {
+		r[k] = v
 	}
-	s.finished = true
-	s.finishTime = t
-	s.tracer.addFinishedSpan(s)
+	r[ext.SpanName] = s.Name
+	r[ext.ServiceName] = s.Service
+	r[ext.ResourceName] = s.Resource
+	r[ext.SpanType] = s.Type
+	return r
 }
 
-// String implements fmt.Stringer.
-func (s *mockspan) String() string {
+func (s *Span) String() string {
 	s.RLock()
 	defer s.RUnlock()
-	sc := s.context
+	sc := s.Context()
+	baggage := make(map[string]string)
+	sc.ForeachBaggageItem(func(k, v string) bool {
+		baggage[k] = v
+		return true
+	})
+
 	return fmt.Sprintf(`
 name: %s
 tags: %#v
 start: %s
-finish: %s
+duration: %s
 id: %d
 parent: %d
 trace: %d
 baggage: %#v
-`, s.name, s.tags, s.startTime, s.finishTime, sc.spanID, s.parentID, sc.traceID, sc.baggage)
+`, s.Name, s.Tags(), s.StartTime(), s.Duration(), sc.SpanID(), s.ParentID(), sc.TraceID(), baggage)
 }
 
-// Context returns the SpanContext of this Span.
-func (s *mockspan) Context() ddtrace.SpanContext { return s.context }
+func (s *Span) ParentID() uint64 {
+	return s.Span.ParentID
+}
 
-// SetUser associates user information to the current trace which the
-// provided span belongs to. The options can be used to tune which user
-// bit of information gets monitored. This mockup only sets the user
-// information as span tags of the root span of the current trace.
-func (s *mockspan) SetUser(id string, opts ...tracer.UserMonitoringOption) {
-	root := s.Root()
-	if root == nil {
-		return
-	}
+func (s *Span) SpanID() uint64 {
+	return s.Span.SpanID
+}
 
-	var cfg tracer.UserMonitoringConfig
-	for _, fn := range opts {
-		fn(&cfg)
-	}
+func (s *Span) TraceID() uint64 {
+	return s.Span.TraceID
+}
 
-	root.SetTag("usr.id", id)
-	root.SetTag("usr.email", cfg.Email)
-	root.SetTag("usr.name", cfg.Name)
-	root.SetTag("usr.role", cfg.Role)
-	root.SetTag("usr.scope", cfg.Scope)
-	root.SetTag("usr.session_id", cfg.SessionID)
+func (s *Span) StartTime() time.Time {
+	return time.Unix(0, s.Span.Start)
 }
 
-// Root walks the span up to the root parent span and returns it.
-// This method is required by some internal packages such as appsec.
-func (s *mockspan) Root() tracer.Span {
-	openSpans := s.tracer.openSpans
-	var current Span = s
-	for {
-		pid := current.ParentID()
-		if pid == 0 {
-			break
-		}
-		parent, ok := openSpans[pid]
-		if !ok {
-			break
-		}
-		current = parent
-	}
-	root, _ := current.(*mockspan)
-	return root
+func (s *Span) Duration() time.Duration {
+	return time.Duration(s.Span.Duration)
+}
+
+func (s *Span) FinishTime() time.Time {
+	return s.StartTime().Add(s.Duration())
 }
 
-// StartChild starts a new child span with the given operation name and options.
-func (s *mockspan) StartChild(operationName string, opts ...ddtrace.StartSpanOption) ddtrace.Span {
-	opts = append(opts, tracer.ChildOf(s.Context()))
-	return s.tracer.StartSpan(operationName, opts...)
+func (s *Span) RealSpan() *tracer.Span {
```

Comment:
I'd have called this `Unwrap` to mimic other similar APIs that exist in other places, but that's also a decent name.

Reply:
`Unwrap` is probably more like what people expect. I'll change this.